### PR TITLE
Find JSON anywhere in message

### PR DESCRIPTION
**Pull request by petrohi**
Since CloudWatch supports this --query should support as well.

### CODE REVIEW

File: awslogs/core.py [hunk 1]
```diff
@@ -175,8 +175,9 @@ def consumer():
                     )
 
                 message = event['message']
-                if self.query is not None and message[0] == '{':
-                    parsed = json.loads(event['message'])
+                json_start = message.find('{')
+                if self.query is not None and json_start != -1:
+                    parsed = json.loads(message[json_start:])
```
Comment (Code0x58) [inline awslogs/core.py]: I think this is a little more optimistic than it was before, and just as far from robust. Is there a reasonable solution to stop possibly unexpected messages from flowing up the script? (I'm assuming it blows up currently).

_Tangent/something of curiosity for me:_ could this code/processing be offloaded into boto3/AWS by using [filter_log_events(…)](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/logs.html#CloudWatchLogs.Client.filter_log_events)?

Comment (jorgebastida) [reply awslogs/core.py]: I agree with @Code0x58 , this is way too optimistic.

In terms of offloading this to aws, the only option they provide to filter (afaik) is already present in ``--filter-pattern``, anything els e hast to be done in the client.

**Comment by jorgebastida** [comment]
I'm afraid I'm going to mark this PR as closed. I don't think it is a good pattern to log both json and plain text at the same time, so the use case feels a bit way too particular to me.

**Comment by petrohi** [comment]
Agreed.

**Pull request closed without merging**